server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -0,0 +1,392 @@
+                .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"))
+            .body("taskId", is(taskId))
+            .body("type", is(DeletedMessageVaultRestoreTask.TYPE))
+            .body("additionalInformation.successfulRestoreCount", is(0))
+            .body("additionalInformation.errorRestoreCount", is(0))
+            .body("additionalInformation.failRemovingCount", is(0))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()));
+    }
+
+    @Test
+    void restoreShouldProduceFailedTaskWithFailsRemovingCountWhenTheVaultDeleteGetsError() {
+        vault.append(USER, DELETED_MESSAGE).block();
+        vault.append(USER, DELETED_MESSAGE_2).block();
+
+        doThrow(new RuntimeException("mock exception"))
+            .when(vault)
+            .delete(any(), any());
+
+        String taskId = given()
+            .when()
+                .post(USER.asString())
+                .jsonPath()
+                .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"))
+            .body("taskId", is(taskId))
+            .body("type", is(DeletedMessageVaultRestoreTask.TYPE))
+            .body("additionalInformation.successfulRestoreCount", is(2))
+            .body("additionalInformation.errorRestoreCount", is(0))
+            .body("additionalInformation.failRemovingCount", is(2))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()));
+    }
+
+    @Test
+    void restoreShouldProduceFailedTaskWhenMailboxMangerGetsError() throws Exception {
+        vault.append(USER, DELETED_MESSAGE).block();
+        vault.append(USER, DELETED_MESSAGE_2).block();
+
+        doThrow(new RuntimeException("mock exception"))
+            .when(mailboxManager)
+            .createMailbox(any(MailboxPath.class), any(MailboxSession.class));
+
+        String taskId = given()
+            .when()
+                .post(USER.asString())
+                .jsonPath()
+                .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"))
+            .body("taskId", is(taskId))
+            .body("type", is(DeletedMessageVaultRestoreTask.TYPE))
+            .body("additionalInformation.successfulRestoreCount", is(0))
+            .body("additionalInformation.errorRestoreCount", is(0))
+            .body("additionalInformation.failRemovingCount", is(0))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()));
+    }
+
+    @Test
+    void restoreShouldRemoveAllMessageInTheVaultOfCorrespondingUser() {
```

Comment:
Can we add a test:

 - MessageManager::append is failing
 - We trigger a restore and wait for its completion
 - All the messages are still in the vault

Reply:
added: `restoreShouldKeepMessagesInTheVaultWhenMessageAppendGetsError`